apps/frontend/src/libs/components/checkbox/checkbox.tsx

```diff
@@ -1,73 +1,127 @@
+import { ZERO_INDEX } from "~/libs/constants/constants.js";
+import { getValidClassNames } from "~/libs/helpers/helpers.js";
 import { useCallback, useFormController } from "~/libs/hooks/hooks.js";
 import {
-	type Control,
-	type FieldPath,
 	type FieldValues,
+	type FormFieldProperties,
 	type InputOption,
 } from "~/libs/types/types.js";
 
 import styles from "./styles.module.css";
 
 type Properties<T extends FieldValues> = {
-	control: Control<T, null>;
 	label: string;
-	name: FieldPath<T>;
+	onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
 	options: InputOption[];
-};
+	variant: "general" | "rounded";
+} & FormFieldProperties<T>;
 
 const Checkbox = <T extends FieldValues>({
 	control,
 	label,
 	name,
+	onChange,
 	options,
+	variant,
 }: Properties<T>): JSX.Element => {
 	const { field } = useFormController({ control, name });
-	const { onChange, value } = field;
+	const { onChange: onFieldChange, value } = field;
+	const isGeneralCheckbox = variant === "general";
 
 	const handleCheckboxesChange = useCallback(
 		(event: React.ChangeEvent<HTMLInputElement>): void => {
 			const isChecked = event.target.checked;
-			const inputValue = Number(event.target.value);
+			const inputValue = isGeneralCheckbox
+				? Number(event.target.value)
+				: event.target.value;
 
 			if (isChecked) {
-				onChange([...value, inputValue]);
+				onFieldChange([...value, inputValue]);
 			} else {
-				onChange((value as number[]).filter((value) => value !== inputValue));
+				onFieldChange(
+					(value as (number | string)[]).filter(
+						(value) => value !== inputValue,
+					),
+				);
 			}
+
+			onChange?.(event);
+		},
+		[isGeneralCheckbox, onFieldChange, onChange, value],
+	);
+
+	const handleSingleCheckboxChange = useCallback(
+		(event: React.ChangeEvent<HTMLInputElement>): void => {
+			onFieldChange(event);
+			onChange?.(event);
 		},
-		[onChange, value],
+		[onFieldChange, onChange],
 	);
 
 	return (
-		<label className={styles["input-wrapper"]}>
-			<p className={styles["question"]}>{label}</p>
-			<div className={styles["input-container"]}>
-				{options.map((option) => {
-					const isChecked = (value as number[]).includes(Number(option.value));
+		<>
+			<label className={styles["input-wrapper"]}>
+				{isGeneralCheckbox && <p className={styles["question"]}>{label}</p>}
```

Comment:
Label should always be present. If you want to hide it, use `hasVisuallyHiddenLabel` prop and `visually-hidden` class